File: Ex-25/ex-25-c++/CalcInterface.cpp
#include "CalcInterface.hpp"

// Inicializamos o vetor com as operacoes válidas
const vector<char> CalcInterface::OPERACOES_VALIDAS = vector<char>({'+', '-', '*', '/', 's'});

CalcInterface::CalcInterface(){

};

CalcInterface::~CalcInterface(){

};

double CalcInterface::recebeOperando(const int &i) const
{
  double operando;
  cout << "Insira o operando " << i << endl;
  cin >> operando;
  return operando;
};

char CalcInterface::recebeOperacao() const
{
  char operacao;
  do
  {
    cout << "Insira uma operacao:" << endl;
    cin >> operacao;
  } while (!CalcInterface::isOperacaoValida(operacao));
  return operacao;
};

void CalcInterface::mostraResultado(const double &resultado) const
{
  cout << "Resultado: " << resultado << endl;
};

bool CalcInterface::isOperacaoValida(const char &operacao)
{
  int size = CalcInterface::OPERACOES_VALIDAS.size();
  for (int i = 0; i < size; i++)
    if (CalcInterface::OPERACOES_VALIDAS[i] == operacao)
      return true;
  return false;
}

void CalcInterface::mostraBoasVindas() const
{
  cout << "--------------------------------" << endl;
  cout << "   Bem-vindo à Calculadora OO   " << endl;
  cout << "--------------------------------" << endl;
}

File: Ex-25/ex_25_java/CalcInterface.java
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class CalcInterface {
    // Inicializamos o vetor com as operacoes válidas
    private static final List<Character> OPERACOES_VALIDAS = Arrays.asList('+', '-', '*', '/', 's');
    private static Scanner scanner = new Scanner(System.in);

    public CalcInterface() {
    }

    public double recebeOperando(int i) {
        System.out.println("Insira o operando " + i);
        double operando = scanner.nextDouble();
        return operando;
    }

    public char recebeOperacao() {
        char operacao;
        do {
            System.out.println("Insira uma operacao:");
            operacao = scanner.next().charAt(0);
        } while (!CalcInterface.isOperacaoValida(operacao));
        return operacao;
    }

    public void mostraResultado(double resultado) {
        System.out.println("Resultado: " + resultado);
    }

    private static boolean isOperacaoValida(char operacao) {
        int size = CalcInterface.OPERACOES_VALIDAS.size();
        for (int i = 0; i < size; i++)
            if (CalcInterface.OPERACOES_VALIDAS.get(i) == operacao)
                return true;
        return false;
    }

    public void mostraBoasVindas() {
        System.out.println("--------------------------------");
        System.out.println("   Bem-vindo à Calculadora OO   ");
        System.out.println("--------------------------------");
    }
}
